src/app/problem/editor/right/testcases/Testcases.js

```diff
@@ -0,0 +1,71 @@
+import React, {useEffect, useState} from "react";
+import Block from "../Block";
+import useTestcaseList from './usecase';
+import {EditSaveCancelButton} from "../../../commons/EditSaveCancelButton";
+import {EditorButton} from "../../../commons/EditorButton";
+import TestcaseEditor from "./TestcaseEditor";
+import {useProblemEditorContext} from "../../ProblemEditorContext";
+
+const TestCaseList = () => {
+    const {currentProblem} = useProblemEditorContext();
+    const {testcases, setTestcases, addNewTestcase, deleteTestcase} = useTestcaseList();
+
+    useEffect(() => {
+        setTestcases(currentProblem.testcases);
+    });
```

Comment:
Should it be `useEffect(() => { ... }, [currentProblem])`?

Reply:
No, I passed the CI.